File: client-android/serenada-core/src/test/java/app/serenada/core/fakes/FakeAudioController.kt
package app.serenada.core.fakes

import app.serenada.core.call.SessionAudioController

class FakeAudioController : SessionAudioController {
    var activateCalls = 0
        private set
    var deactivateCalls = 0
        private set
    var proximityPausesVideo = false

    override fun activate() { activateCalls++ }
    override fun deactivate() { deactivateCalls++ }

    override fun shouldPauseVideoForProximity(isScreenSharing: Boolean): Boolean =
        proximityPausesVideo
}

File: client-ios/SerenadaCore/Tests/SerenadaCoreTests/Fakes/FakeAudioController.swift
import Foundation
@testable import SerenadaCore

@MainActor
final class FakeAudioController: SessionAudioController {
    private(set) var activateCalls = 0
    private(set) var deactivateCalls = 0
    var proximityPausesVideo = false
    private var audioEnvironmentChangedHandler: (() -> Void)?

    func activate() { activateCalls += 1 }
    func deactivate() { deactivateCalls += 1 }

    func shouldPauseVideoForProximity(isScreenSharing: Bool) -> Bool {
        proximityPausesVideo
    }

    func setOnAudioEnvironmentChanged(_ handler: @escaping () -> Void) {
        audioEnvironmentChangedHandler = handler
    }

    func triggerAudioEnvironmentChanged() {
        audioEnvironmentChangedHandler?()
    }
}
